src/openlcb/BroadcastTime.hxx

```diff
@@ -0,0 +1,448 @@
+/** @copyright
+ * Copyright (c) 2018, Stuart W. Baker
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ *
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file BroadcastTime.hxx
+ *
+ * Implementation of a Broadcast Time Protocol Interface.
+ *
+ * @author Stuart W. Baker
+ * @date 4 November 2018
+ */
+
+#ifndef _OPENLCB_BROADCASTTIME_HXX_
+#define _OPENLCB_BROADCASTTIME_HXX_
+
+#include <functional>
+
+#include "openlcb/BroadcastTimeDefs.hxx"
+#include "openlcb/EventHandlerTemplates.hxx"
+
+namespace openlcb
+{
+
+/// Implementation of a Broadcast Time Protocol Interface.
+class BroadcastTime : public SimpleEventHandler
+                    , public StateFlowBase
+                    , protected Atomic
+{
+public:
+    /// Constructor.
```

Comment:
Is this supposed to be public or rather protected? i.e. is it expected that applications are creating these directly (to operate as a time client)?